### ISSUE TITLE
 TOML: using to_toml option may produce incorrect TOML

### ISSUE THREAD

**Issue by stephenll** [JuliaLang/julia#46454]
Here is is a simplified example to illustrate the issue I'm seeing with using the TOML writer with the `to_toml` option for writing a custom type.  It seems that when using the to_toml option,  TOML print doesn't print the correct TOML.

I expect the `hello = "world"` to not be combined with the `[b]` table.

I'm using Julia 1.8 on both a M1 Mac (arm64) and an Intel Mac.

Just looking to see if anyone else is seeing this and if my understanding is correct.

```
using TOML

struct Foo
    a::Int64
    b::Float64
end

struct Bar
    c::Float64
    d::String
end


f = Foo(2,9.9)
b = Bar(1.345, "hello")

dd = Dict("hello"=>"world", "f"=>f,  "b"=>b)

toDict(foo::Foo) = Dict("a"=>foo.a, "b"=>foo.b)
toDict(bar::Bar) = Dict("c"=>bar.c, "d"=>bar.d)

TOML.print(toDict,dd)
```
which results in the "hello" being added to the `b` table in the TOML.
```
[f]
b = 9.9
a = 2

[b]
c = 1.345
d = "hello"
hello = "world"
```

**Comment by vtjnash** [comment]
Added in https://github.com/JuliaLang/julia/pull/41009 (@aviatesk). Looks like the added logic didn't follow the comment on the loop here that is_tabular data shouldn't be printed in this pass

**Comment by stephenll** [comment]
In anyone is following and needs a temporary fix... I think I found a work around via custom sorting. First, create a vector of strings of fields that are non-tabular.  Next, create a function to see if x is not in the list.  We want false assigned to those in the list, so when it comes to sorting, those are forced first since they are effectively 0. Here is a snippet:

```julia
list = ["hello"]
TOML.print(toDict, dd, sorted=true, by = x->x ∉ list)
```

So far it works on the few test cases I tried.
